File: sol/881/881.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        res = 0
        i = 0
        j = len(people) - 1
        while i < j:
            if people[i] + people[j] <= limit:
                i += 1
            j -= 1
            res += 1
        if i == j:
            res += 1
        return res

File: sol/881/881.cpp
struct Solution {
    inline int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int i = 0 , j = people.size() - 1, res = 0;
        while (i < j){
            if (people[i] + people[j--] <= limit){
                ++i;
            }
            ++res;
        }
        if (i == j){
            ++res;
        }
        return res;
    }
};
